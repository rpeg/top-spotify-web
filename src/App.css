.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

body {
  background-color: #191414 !important;
  color: #FFFFFF !important;
  font-family: 'DM Sans', sans-serif !important;
  margin-top: 3em !important;
}