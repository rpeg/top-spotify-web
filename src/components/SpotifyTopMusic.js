import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { findKey } from 'lodash';
import {
  Spinner, Container, Row, Col,
} from 'react-bootstrap';

import ArtistGrid from './ArtistGrid';
import TrackGrid from './TrackGrid';
import WordCloud from './WordCloud';
import Features from './Features';
import ComponentHeader from './ComponentHeader';
import {
  receiveArtists,
  receiveTracks,
  receiveFeatures,
  fetchArtistsIfNeeded,
  fetchTracksIfNeeded,
  fetchFeaturesIfNeeded,
} from '../actions/actions';
import { TimeRanges } from '../constants/constants';

const SpotifyTopMusic = ({ socket }) => {
  const user = useSelector((state) => state.user.user);
  const timeRange = useSelector((state) => state.timeRange);
  const artistCount = useSelector((state) => state.artistCount);
  const trackCount = useSelector((state) => state.trackCount);
  const genreCount = useSelector((state) => state.genreCount);
  const artists = useSelector((state) => state.artists);
  const tracks = useSelector((state) => state.tracks);
  const features = useSelector((state) => state.features);
  const dispatch = useDispatch();

  useEffect(() => {
    if (user && user.id) {
      dispatch(fetchArtistsIfNeeded(timeRange, socket.id));
      dispatch(fetchTracksIfNeeded(timeRange, socket.id));
    }
  }, [user]);

  useEffect(() => {
    socket.on('topArtists', (result) => {
      const matchingTimeRange = findKey(TimeRanges, (r) => r.range === result.range);
      dispatch(receiveArtists(matchingTimeRange || TimeRanges.LONG, result.items));
    });
  });

  useEffect(() => {
    socket.on('topTracks', (result) => {
      const matchingTimeRange = findKey(TimeRanges, (r) => r.range === result.range);
      dispatch(receiveTracks(matchingTimeRange || TimeRanges.LONG, result.items));

      // Derive features from tracks
      // dispatch(fetchFeaturesIfNeeded(matchingTimeRange, socket.id));
    });
  });

  useEffect(() => {
    socket.on('topFeatures', (result) => {
      const matchingTimeRange = findKey(TimeRanges, (r) => r.range === result.range);
      dispatch(receiveFeatures(matchingTimeRange || TimeRanges.LONG, result.items));
    });
  });

  return (
    user && user.id ? (
      <div style={{ marginTop: '2em' }}>
        <div>
          <h1>{TimeRanges[timeRange]}</h1>
        </div>
        <Container>
          <Row>
            <ComponentHeader title="Artists" />
          </Row>
          {artists && artists.items && artists.items.length ? (
            <ArtistGrid artists={artists.items.slice(0, artistCount)} numRows={2} />
          ) : <Spinner animation="border" />}
          <Row style={{ paddingTop: '2em' }}>
            <Col xs={8} style={{ padding: '0 0.5em 0 0' }}>
              {artists.length ? (
                <WordCloud
                  genres={[...artists.map((a) => a.genres).flat()]}
                  count={genreCount}
                />
              ) : <Spinner animation="border" />}
              {features.length ? (<Features features={features} />) : <Spinner animation="border" />}
            </Col>
            <Col xs={4} style={{ padding: '0 0 0 0.5em' }}>
              {tracks && tracks.items && tracks.items.length ? (<TrackGrid tracks={tracks.items} count={trackCount} />) : <Spinner animation="border" />}
            </Col>
          </Row>
        </Container>
      </div>
    ) : null
  );
};

SpotifyTopMusic.propTypes = {
  socket: PropTypes.object.isRequired,
};

export default SpotifyTopMusic;
