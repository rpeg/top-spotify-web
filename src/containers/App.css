@font-face {
  font-family: 'Karla';
  font-style: normal;
  font-weight: 400;
  src: local('Karla'), local('Karla-Regular'), url(https://fonts.gstatic.com/s/karla/v8/qkBbXvYC6trAT7RVLtyU5rZP.woff2) format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}

@font-face {
  font-family: 'Karla';
  font-style: normal;
  font-weight: 700;
  src: local('Karla Bold'), local('Karla-Bold'), url(https://fonts.gstatic.com/s/karla/v8/qkBWXvYC6trAT7zuC8m5xLtlmgzD.woff2) format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

body, #top-spotify {
  background-color: #191414 !important;
  color: #FFFFFF !important;
  font-family: 'Karla', sans-serif !important;
  margin-top: 3em !important;
}

.splash {
  width: 90%
}

.splash-header {
  position: fixed;
  z-index: 25;
}

.splash-title {
  left: 10vw;
  text-align: left;
  margin: 6vh 10vw 0 9.75vw;
  font-size: 6vw;
  font-weight: bold;
}

.splash-desc {
  text-align: left;
  opacity: .9;
  margin: 2vh 10vw 0 10vw;
  font-size: 3vh;
}

.splash-example {
  position: fixed;
  top: 35%;
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: 100%;
  background-position: center top;
}

.xs, .s {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.btn-primary {
  color: #fff !important;
  background-color: rgba(29, 185, 84, 1) !important;
  border-color: rgba(29, 185, 84, 1) !important;
}

.btn-primary:hover {
  color: #fff !important;
  border-color: rgba(29, 185, 84, 0.75) !important;
  background-color: rgba(29, 185, 84, 0.75) !important;
}

.grid-row li:not(:first-child) > div {
  margin-top: 1em;
}

.stat-content {
  text-align: start;
}

.stat-content p {
  margin: 0;
  line-height: 1;
  font-size: 32pt;
}

.stat-title p {
  font-weight: bold;
  font-size: 18pt;
  margin: 1em 0 0.4em 0;
}