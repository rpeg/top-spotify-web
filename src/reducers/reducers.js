import { combineReducers } from 'redux';
import {
  SET_USER,
  SET_TIME_RANGE,
  TimeRanges,
  REQUEST_ARTISTS,
  RECEIVE_ARTISTS,
  REQUEST_TRACKS,
  RECEIVE_TRACKS,
  REQUEST_FEATURES,
  RECEIVE_FEATURES,
  SET_ARTIST_COUNT,
  SET_TRACK_COUNT,
  SET_GENRE_COUNT,
} from '../constants/constants';

const { LONG } = TimeRanges;

function user(state = {}, action) {
  switch (action.type) {
    case SET_USER:
      return { ...state, user: action.user };
    default:
      return state;
  }
}

function timeRange(state = LONG, action) {
  switch (action.type) {
    case SET_TIME_RANGE:
      return { ...state, timeRange: action.timeRange };
    default:
      return state;
  }
}

function artistCount(state = 10, action) {
  switch (action.type) {
    case SET_ARTIST_COUNT:
      return { ...state, artistCount: action.artistCount };
    default:
      return state;
  }
}

function trackCount(state = 15, action) {
  switch (action.type) {
    case SET_TRACK_COUNT:
      return { ...state, trackCount: action.trackCount };
    default:
      return state;
  }
}

function genreCount(state = 40, action) {
  switch (action.type) {
    case SET_GENRE_COUNT:
      return { ...state, genreCount: action.genreCount };
    default:
      return state;
  }
}

function artists(state = {}, action) {
  switch (action.type) {
    case REQUEST_ARTISTS:
      return { ...state, isFetching: true };
    case RECEIVE_ARTISTS:
      return { ...state, isFetching: false, items: action.items };
    default:
      return state;
  }
}

function artistsByTimeRange(state = {}, action) {
  switch (action.type) {
    case RECEIVE_ARTISTS:
    case REQUEST_ARTISTS:
      return { ...state, [action.timeRange]: artists(state[action.timeRange], action) };
    default:
      return state;
  }
}

function tracks(state = {}, action) {
  switch (action.type) {
    case REQUEST_TRACKS:
      return { ...state, isFetching: true };
    case RECEIVE_TRACKS:
      return { ...state, isFetching: false, items: action.tracks };
    default:
      return state;
  }
}

function tracksByTimeRange(state = {}, action) {
  switch (action.type) {
    case RECEIVE_TRACKS:
    case REQUEST_TRACKS:
      return { ...state, [action.timeRange]: tracks(state[action.timeRange], action) };
    default:
      return state;
  }
}

function features(state = {}, action) {
  switch (action.type) {
    case REQUEST_FEATURES:
      return { ...state, isFetching: true };
    case RECEIVE_FEATURES:
      return { ...state, isFetching: false, items: action.features };
    default:
      return state;
  }
}

const rootReducer = combineReducers({
  user,
  timeRange,
  artistCount,
  trackCount,
  genreCount,
  artists,
  artistsByTimeRange,
  tracks,
  tracksByTimeRange,
  features,
});

export default rootReducer;
