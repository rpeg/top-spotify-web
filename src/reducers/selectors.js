export const hasClickedCreate = state => state.hasClickedCreate;
