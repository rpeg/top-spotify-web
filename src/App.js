import React, { useState } from "react";
import axios from "axios";
import { Button } from "react-bootstrap";
import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";

import ArtistGrid from "./components/ArtistGrid";
import DonutChart from "./components/DonutChart";

const App = () => {
  const [artistsLongTerm, setArtistsLongTerm] = useState([]);
  const [artistsMedTerm, setArtistsMedTerm] = useState([]);
  const [artistsShortTerm, setArtistsShortTerm] = useState([]);

  const loginSpotify = e => {
    e.preventDefault();
    axios
      .get("http://localhost:3000/login")
      .then(res => (window.location.href = res.data))
      .catch(err => console.log(err));
  };

  // TODO multithreaded
  const getData = e => {
    e.preventDefault();
    getArtistsLongTerm();
    getArtistsMedTerm();
    getArtistsShortTerm();
  };

  const getArtistsLongTerm = () => {
    axios
      .get("http://localhost:3000/my-top-artists", {
        params: {
          time_range: "long_term",
          limit: 50
        }
      })
      .then(res => setArtistsLongTerm(res.data.items))
      .catch(err => console.log(err));
  };

  const getArtistsMedTerm = () => {
    axios
      .get("http://localhost:3000/my-top-artists", {
        params: {
          time_range: "medium_term",
          limit: 50
        }
      })
      .then(res => setArtistsMedTerm(res.data.items))
      .catch(err => console.log(err));
  };

  const getArtistsShortTerm = () => {
    axios
      .get("http://localhost:3000/my-top-artists", {
        params: {
          time_range: "short_term",
          limit: 50
        }
      })
      .then(res => setArtistsShortTerm(res.data.items))
      .catch(err => console.log(err));
  };

  return (
    <div className="App">
      <Button variant="outline-primary" onClick={loginSpotify}>
        Login
      </Button>
      <Button variant="outline-primary" onClick={getData}>
        Get data
      </Button>
      {artistsLongTerm.length > 0 && artistsMedTerm.length > 0 && (
        <DonutChart
          genres={[
            ...artistsLongTerm.map(a => a.genres).flat(),
            ...artistsMedTerm.map(a => a.genres).flat()
          ]}
          count={10}
        />
      )}
      <div>
        {artistsLongTerm.length > 0 && (
          <ArtistGrid
            artists={artistsLongTerm.slice(0, 8)}
            title="All time"
            numRows={2}
          />
        )}
        {artistsMedTerm.length > 0 && (
          <ArtistGrid
            artists={artistsMedTerm.slice(0, 8)}
            title="Past 6 months"
            numRows={2}
          />
        )}
        {artistsShortTerm.length > 0 && (
          <ArtistGrid
            artists={artistsShortTerm.slice(0, 8)}
            title="Recently"
            numRows={2}
          />
        )}
      </div>
    </div>
  );
};

export default App;
