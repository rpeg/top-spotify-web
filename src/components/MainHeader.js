import React from 'react';
import { useSelector } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap';
import Moment from 'moment';

import Profile from './Profile';
import { selectUser } from '../reducers/selectors';

const MainHeader = () => {
  const user = useSelector(selectUser);

  return (
    user && user.id ? (
      <Container style={{ paddingTop: '2em' }}>
        <Row>
          <Col xs={6} style={{ padding: '0', display: 'block' }}>
            <span
              style={{
                display: 'inline-flex',
                alignItems: 'center',
                float: 'left',
              }}
            >
              <img
                style={{ height: '50px', width: 'auto' }}
                src={`${process.env.PUBLIC_URL}/Spotify_Icon_RGB_Green.png`}
                alt="Spotify Logo"
              />
              <div style={{ width: 'auto', margin: '0 0 0 0.5em' }}>
                <p className="text-m" style={{ textAlign: 'left', margin: '0' }}><b>My Top Music</b></p>
                <p className="text-s" style={{ textAlign: 'left', margin: '0' }}>
                  {Moment().format('MMM. Do, YYYY')}
                </p>
              </div>
            </span>
          </Col>
          <Col xs={6} style={{ padding: '0', display: 'block' }}>
            <Profile user={user} />
          </Col>
        </Row>
      </Container>
    ) : null
  );
};

export default MainHeader;
