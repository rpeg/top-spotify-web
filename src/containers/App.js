import React from 'react';
import { useSelector } from 'react-redux';
import io from 'socket.io-client';

import OAuth from '../components/OAuth';
import API_URL from '../config';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import MainHeader from '../components/MainHeader';
import SpotifyTopMusic from '../components/SpotifyTopMusic';

import Controls from '../components/Controls';
import Splash from '../components/Splash';
import { selectHasClickedCreate, selectUser } from '../reducers/selectors';

const socket = io(API_URL);

const App = () => {
  const hasClickedCreate = useSelector(selectHasClickedCreate);
  const user = useSelector(selectUser);

  return (
    <div className="App">
      <OAuth socket={socket} />
      <Controls />
      {hasClickedCreate && (
        <div id="top-spotify">
          <MainHeader />
          <SpotifyTopMusic />
        </div>
      )}
      {!user.id && (
        <Splash />
      )}
    </div>
  );
};

export default App;
