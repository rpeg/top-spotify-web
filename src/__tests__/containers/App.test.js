import React from 'react';
import { shallow } from 'enzyme';

import App from '../../containers/App';
import MainHeader from '../../components/MainHeader';
import SpotifyTopMusic from '../../components/SpotifyTopMusic';
import * as selectors from '../../reducers/selectors'

jest.mock("react-redux", () => ({
  useSelector: jest.fn(fn => fn()),
  useDispatch: () => jest.fn()
}));

const setup = ({ hasClickedCreate = false }) => {
  jest.spyOn(selectors, "hasClickedCreate").mockReturnValue(hasClickedCreate);
};

describe('App', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should not log errors in console', () => {
    setup({ hasClickedCreate: false });
    const wrapper = shallow(<App />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('initial renders correctly', () => {
    setup({ hasClickedCreate: false });
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(MainHeader).exists()).toBe(false);
    expect(wrapper.find(SpotifyTopMusic).exists()).toBe(false);
  });

  describe('conditional renders', () => {
    it('renders MainHeader and SpotifyTopMusic if hasClickedCreate', () => {
      setup({ hasClickedCreate: true });
      const wrapper = shallow((<App />));
      expect(wrapper.find(MainHeader).exists()).toBe(true);
      expect(wrapper.find(SpotifyTopMusic).exists()).toBe(true);
    })
  })
});
