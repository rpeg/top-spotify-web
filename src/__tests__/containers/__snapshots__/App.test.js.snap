// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App initial renders correctly 1`] = `
<div
  className="App"
>
  <OAuth
    socket={
      Socket {
        "_callbacks": Object {
          "$connect": Array [
            [Function],
          ],
          "$connecting": Array [
            [Function],
          ],
        },
        "acks": Object {},
        "connected": false,
        "disconnected": true,
        "flags": Object {},
        "ids": 0,
        "io": Manager {
          "_callbacks": Object {
            "$close": Array [
              [Function],
            ],
            "$open": Array [
              [Function],
            ],
            "$packet": Array [
              [Function],
            ],
          },
          "_randomizationFactor": 0.5,
          "_reconnection": true,
          "_reconnectionAttempts": Infinity,
          "_reconnectionDelay": 1000,
          "_reconnectionDelayMax": 5000,
          "_timeout": 20000,
          "autoConnect": true,
          "backoff": Backoff {
            "attempts": 0,
            "factor": 2,
            "jitter": 0.5,
            "max": 5000,
            "ms": 1000,
          },
          "connecting": Array [
            [Circular],
          ],
          "decoder": Decoder {
            "reconstructor": null,
          },
          "encoder": Encoder {},
          "encoding": false,
          "engine": Socket {
            "_callbacks": Object {
              "$error": Array [
                [Function],
              ],
              "$open": Array [
                [Function],
              ],
            },
            "agent": false,
            "binaryType": null,
            "ca": null,
            "cert": null,
            "ciphers": null,
            "enablesXDR": false,
            "forceBase64": false,
            "forceJSONP": false,
            "forceNode": false,
            "hostname": "127.0.0.1",
            "id": null,
            "isReactNative": false,
            "jsonp": true,
            "key": null,
            "onlyBinaryUpgrades": undefined,
            "passphrase": null,
            "path": "/socket.io/",
            "perMessageDeflate": Object {
              "threshold": 1024,
            },
            "pfx": null,
            "pingInterval": null,
            "pingIntervalTimer": null,
            "pingTimeout": null,
            "pingTimeoutTimer": null,
            "policyPort": 843,
            "port": "3000",
            "prevBufferLen": 0,
            "query": Object {},
            "readyState": "opening",
            "rejectUnauthorized": true,
            "rememberUpgrade": false,
            "secure": false,
            "timestampParam": "t",
            "timestampRequests": undefined,
            "transport": XHR {
              "_callbacks": Object {
                "$close": Array [
                  [Function],
                ],
                "$drain": Array [
                  [Function],
                ],
                "$error": Array [
                  [Function],
                ],
                "$packet": Array [
                  [Function],
                ],
              },
              "agent": false,
              "ca": null,
              "cert": null,
              "ciphers": null,
              "enablesXDR": false,
              "extraHeaders": undefined,
              "forceNode": false,
              "hostname": "127.0.0.1",
              "isReactNative": false,
              "key": null,
              "localAddress": undefined,
              "passphrase": null,
              "path": "/socket.io/",
              "pfx": null,
              "pollXhr": Request {
                "_callbacks": Object {
                  "$data": Array [
                    [Function],
                  ],
                  "$error": Array [
                    [Function],
                  ],
                },
                "agent": false,
                "async": true,
                "ca": null,
                "cert": null,
                "ciphers": null,
                "data": null,
                "enablesXDR": false,
                "extraHeaders": undefined,
                "index": 0,
                "isBinary": undefined,
                "key": null,
                "method": "GET",
                "passphrase": null,
                "pfx": null,
                "rejectUnauthorized": true,
                "requestTimeout": undefined,
                "supportsBinary": false,
                "uri": "http://127.0.0.1:3000/socket.io/?EIO=3&transport=polling&t=MxDNrtb&b64=1",
                "withCredentials": true,
                "xd": true,
                "xhr": XMLHttpRequest {
                  "DONE": 4,
                  "HEADERS_RECEIVED": 2,
                  "LOADING": 3,
                  "OPENED": 1,
                  "UNSENT": 0,
                  "abort": [Function],
                  "addEventListener": [Function],
                  "dispatchEvent": [Function],
                  "getAllResponseHeaders": [Function],
                  "getRequestHeader": [Function],
                  "getResponseHeader": [Function],
                  "handleError": [Function],
                  "onreadystatechange": [Function],
                  "open": [Function],
                  "readyState": 1,
                  "removeEventListener": [Function],
                  "responseText": "",
                  "responseXML": "",
                  "send": [Function],
                  "setDisableHeaderCheck": [Function],
                  "setRequestHeader": [Function],
                  "status": null,
                  "statusText": null,
                },
                "xs": false,
              },
              "polling": true,
              "port": "3000",
              "query": Object {
                "EIO": 3,
                "b64": 1,
                "t": "MxDNrtb",
                "transport": "polling",
              },
              "readyState": "opening",
              "rejectUnauthorized": true,
              "requestTimeout": undefined,
              "secure": false,
              "socket": [Circular],
              "supportsBinary": false,
              "timestampParam": "t",
              "timestampRequests": undefined,
              "withCredentials": true,
              "xd": true,
              "xs": false,
            },
            "transportOptions": Object {},
            "transports": Array [
              "polling",
              "websocket",
            ],
            "upgrade": true,
            "upgrades": null,
            "withCredentials": true,
            "writeBuffer": Array [],
          },
          "lastPing": null,
          "nsps": Object {
            "/": [Circular],
          },
          "opts": Object {
            "hostname": "127.0.0.1",
            "path": "/socket.io",
            "port": "3000",
            "secure": false,
          },
          "packetBuffer": Array [],
          "readyState": "opening",
          "skipReconnect": false,
          "subs": Array [
            Object {
              "destroy": [Function],
            },
            Object {
              "destroy": [Function],
            },
            Object {
              "destroy": [Function],
            },
          ],
          "uri": "http://127.0.0.1:3000",
        },
        "json": [Circular],
        "nsp": "/",
        "receiveBuffer": Array [],
        "sendBuffer": Array [],
        "subs": Array [
          Object {
            "destroy": [Function],
          },
          Object {
            "destroy": [Function],
          },
          Object {
            "destroy": [Function],
          },
        ],
      }
    }
  />
  <Controls />
</div>
`;
